src/antsibull/build_changelog.py

```diff
@@ -12,27 +12,70 @@
 import os
 import os.path
 import typing as t
+from dataclasses import dataclass
 
 from antsibull_changelog.changelog_generator import (
     ChangelogEntry as ChangelogGeneratorEntry,
 )
-from antsibull_changelog.changelog_generator import ChangelogGenerator
-from antsibull_changelog.config import DEFAULT_SECTIONS
-from antsibull_changelog.rst import RstBuilder
+from antsibull_changelog.config import DEFAULT_SECTIONS, TextFormat
+from antsibull_changelog.rendering.changelog import (
+    add_section_content,
+    create_document_renderer,
+)
+from antsibull_changelog.rendering.document import (
+    AbstractRenderer,
+    DocumentRenderer,
+    SectionRenderer,
+)
+from antsibull_changelog.rendering.rst_document import RSTDocumentRenderer
 from antsibull_core import app_context
+from antsibull_core.logging import log
 from packaging.version import Version as PypiVer
 
 from .changelog import Changelog, ChangelogData, ChangelogEntry, get_changelog
 from .collection_meta import CollectionsMetadata
 from .utils.galaxy import create_galaxy_context
 
+mlog = log.fields(mod=__name__)
+
+
+class SectionAdder:
+    parent: SectionAdder | AbstractRenderer
+    section_name: str
+    section: SectionRenderer | None
+
+    def __init__(self, parent: SectionAdder | AbstractRenderer, section_name: str):
+        self.parent = parent
+        self.section_name = section_name
+        self.section = None
+
+    def get_section(self) -> SectionRenderer:
+        if self.section is None:
+            parent: AbstractRenderer
+            if isinstance(self.parent, SectionAdder):
+                parent = self.parent.get_section()
+            else:
+                parent = t.cast(AbstractRenderer, self.parent)
+            self.section = parent.add_section(self.section_name)
+        return self.section
+
+    def close(self) -> None:
+        if self.section is not None:
+            self.section.close()
+
+
 #
 # Changelog
 #
 
+CHANGELOG_FORMATS = [TextFormat.RESTRUCTURED_TEXT, TextFormat.MARKDOWN]
 
 PluginDataT = t.List[
-    t.Tuple[str, str, ChangelogGenerator, t.Optional[ChangelogGeneratorEntry]]
+    t.Tuple[
+        str,
+        str,
+        t.Optional[ChangelogGeneratorEntry],
```

Comment:
Black should format this more compactly if you remove the trailing `,`.

Reply:
c7aa28e90c67b0541fa130ab265248c93b707a87